blocktx/client.go

```diff
@@ -75,48 +68,37 @@ func NewClient(client blocktx_api.BlockTxAPIClient, opts ...Option) ClientI {
 }
 
 func (btc *Client) Start(minedBlockChan chan *blocktx_api.Block) {
-	btc.retryTicker = time.NewTicker(btc.retryInterval)
+	defer func() {
+		btc.shutdownCompleteStart <- struct{}{}
+	}()
 
-	go func() {
-		defer func() {
-			btc.shutdownCompleteStart <- struct{}{}
-		}()
+	stream, err := btc.client.GetBlockNotificationStream(context.Background(), &blocktx_api.Height{})
+	if err != nil {
+		btc.logger.Error("failed to get block notification stream", slog.String("err", err.Error()))
+		return
+	}
 
+	btc.logger.Info("Connected to block-tx server")
+
+	go func() {
 		for {
 			select {
-			case <-btc.retryTicker.C:
-				stream, err := btc.client.GetBlockNotificationStream(context.Background(), &blocktx_api.Height{})
-				if err != nil {
-					btc.logger.Error("failed to get block notification stream", slog.String("err", err.Error()))
-					continue
-				}
-
-				btc.logger.Info("Connected to block-tx server")
-
-				var block *blocktx_api.Block
-				for {
-					select {
-					case <-btc.shutdown:
-						return
-					default:
-						block, err = stream.Recv()
-						if err != nil {
-							btc.logger.Error("Failed to receive block", slog.String("err", err.Error()))
-							break
-						}
-						btc.logger.Info("Block", slog.String("hash", utils.ReverseAndHexEncodeSlice(block.Hash)))
-						utils.SafeSend(minedBlockChan, block)
-					}
-				}
 			case <-btc.shutdown:
 				return
+			default:
```

Comment:
In the PR https://github.com/bitcoin-sv/arc/pull/108 I didn't want to change the logic, but just add `slog` and make it testable.

This PR changes the logic of the `Start` function. 

The original logic first gets the stream every `btc.retryTicker` interval and then iterates over the stream

Here we don't wait for incoming interval ticks `btc.retryTicker.C`  to first get the stream `btc.client.GetBlockNotificationStream` but the the connection is open all the time.

The question is if we should change the logic just for better readability. Need to check that first